includes/compatibility/civicrm.wpml.php

```diff
@@ -100,4 +106,264 @@ public function locale_filter($locale) {
+      if ($language['active']) {
+        $slug = $id;
+        break;
+      }
+    }
+
+    if (empty($slug)) {
+      return $url;
+    }
+
+    // Obtain lagnuage negotiation setting
+    $wpml_negotiation = apply_filters('wpml_setting', NULL, 'language_negotiation_type');
+
+    // Build the modified URL.
+    global $sitepress;
+    $raw_url = $this->wpml_remove_language_from_url($url, $sitepress, $wpml_negotiation);
+    $language_url = $sitepress->convert_url($raw_url, $slug);
+
+    return $language_url;
+
+  }
+
+  /**
+   * Setup the rewrite rules for WPML.
+   *
+   * @since 5.70
+   *
+   * @param bool $flush_rewrite_rules True if rules flushed, false otherwise.
+   * @param WP_Post $basepage The Base Page post object.
+   */
+  public function rewrite_rules_wpml($flush_rewrite_rules, $basepage) {
+
+    // Bail if WPML is not present.
+    if (!defined('ICL_SITEPRESS_VERSION')) {
+      return;
+    }
+
+    /*
+     * Collect all rewrite rules into an array.
+     *
+     * Because the array of specific Post IDs is added *after* the array of
+     * paths for the Base Page ID, those specific rewrite rules will "win" over
+     * the more general Base Page rules.
+     */
+    global $sitepress;
+    $collected_rewrites = [];
+
+    // Grab information about configuration
+    $wpml_active = apply_filters('wpml_active_languages', NULL);
+
+    // Obtain lagnuage negotiation setting
+    $wpml_negotiation = apply_filters('wpml_setting', NULL, 'language_negotiation_type');
+
+    // Support prefixes for a single Base Page.
+    $basepage_url = get_permalink($basepage->ID);
+    $basepage_raw_url = $this->wpml_remove_language_from_url($basepage_url, $sitepress, $wpml_negotiation);
+    foreach ($wpml_active as $slug => $data) {
+      $language_url = $sitepress->convert_url($basepage_raw_url, $slug);
+      $parsed_url = wp_parse_url($language_url, PHP_URL_PATH);
+      $regex_path = substr($parsed_url, 1);
+
+      // Determine if there is a translation path
+      $post_id = apply_filters('wpml_object_id', $basepage->ID, 'page', FALSE, $slug);
+      if ($post_id == $basepage->ID) {
+        $collected_rewrites[$post_id][] = $regex_path;
+      }
+      else {
+        $url = get_permalink($post_id);
+        $url = $sitepress->convert_url($url, $slug);
+        $parsed_url = wp_parse_url($url, PHP_URL_PATH);
+        $regex_path = substr($parsed_url, 1);
+        $collected_rewrites[$post_id][] = $regex_path;
+      }
+
+    };
+
+    // Make collection unique and add remaining rewrite rules.
+    $rewrites = array_map('array_unique', $collected_rewrites);
+    if (!empty($rewrites)) {
+      foreach ($rewrites as $post_id => $rewrite) {
+        foreach ($rewrite as $path) {
+          add_rewrite_rule(
+            '^' . $path . '([^?]*)?',
+            'index.php?page_id=' . $post_id . '&civiwp=CiviCRM&q=civicrm%2F$matches[1]',
+            'top'
+          );
+        }
+      }
+    }
+    $this->rewrites = $rewrites;
```

Comment:
@shaneonabike You're going to need to declare the dynamic properties otherwise PHP 8.2+ will write lots of warnings to the logs.

Reply:
Oh boy! I thought I had declared that good catch thanks!